pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +554,93 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	if cq.Status.PendingWorkloadsStatus == nil {
+		return &kueue.ClusterQueuePendingWorkloadsStatus{
+			Head:           r.qManager.GetSnapshot(cq.Name),
+			LastChangeTime: metav1.Time{Time: time.Now()},
+		}
+	}
+	if time.Since(cq.Status.PendingWorkloadsStatus.LastChangeTime.Time) >= r.queueVisibilityUpdateInterval {
```

Comment:
I suggest making this one `if` calling `shouldUpdatePendingWorkloadStatus`. 

Then `shouldUpdatePendingWorkloadStatus` would check `cq.Status.PendingWorkloadsStatus == nil` OR 
(`time.Since(cq.Status.PendingWorkloadsStatus.LastChangeTime.Time) >= r.queueVisibilityUpdateInterval` AND  !equality.Semantic.DeepEqual(cq.Status.PendingWorkloadsStatus.Head, pendingWorkloads))`

The benefit is that then we have a single place when we create `kueue.ClusterQueuePendingWorkloadsStatus`.

Reply:
added here https://github.com/kubernetes-sigs/kueue/pull/1069/commits/59446b99221c239140ce79325634fcb5ec9cee27